### ISSUE TITLE
'Legacy' add-ons don't support secrets substitution in configuration options

### ISSUE THREAD

**Issue by goldfoxo** [home-assistant/supervisor#3568]
### Describe the issue you are experiencing

I'm attempting to use a 3rd-party addon (https://github.com/fuatakgun/eufy_security_addon) which has the `"legacy": true` config option, ie the docker container is unaware of Home Assistant, relying only on environment vars for config.

The addon is configured with a username and password, and I'd like to use secret substitution to keep my secrets centralised, as I do with all my other credentials and addons. However, it seems that the environment vars that are set in the legacy container aren't aware of secret substitution. The instead are set to the literal '!secret abc' string instead.

### What is the used version of the Supervisor?

supervisor-2022.04.0

### What type of installation are you running?

Home Assistant Supervised

### Which operating system are you running on?

Debian

### What is the version of your installed operating system?

11

### What version of Home Assistant Core is installed?

core-2022.4.3

### Steps to reproduce the issue

1. In secrets.yaml, add secrets `eufy_username` and `eufy_password` containing my credentials.
2. Add https://github.com/fuatakgun/eufy_security_addon as an Add-on Store repository.
3. Install Eufy Security Add-on.
4. In the Add-on Options, set Username to `!secret eufy_username` and Password to `!secret eufy_password`.
5. Start add-on and observe it fail to authenticate.
6. In the container, the environment vars will be literally `USERNAME=!secret eufy_username` and `PASSWORD=!secret eufy_password` rather than the secret's values.
7. Not using secrets.yaml, and directly configuring the options will work correctly.

### Anything in the Supervisor logs that might be useful for us?

```txt
The logs do mention the secrets being resolved, but I'm assuming that's just for the API, which the legacy addon doesn't use.

22-04-14 16:17:04 INFO (SyncWorker_5) [supervisor.docker.addon] Starting Docker add-on bropat/eufy-security-ws with version 0.8.4
22-04-14 16:17:04 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret eufy_username
22-04-14 16:17:04 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret eufy_password
```

### Additional information

I've taken a look into Supervisor and the simplest fix I've come up with is to change [line 121 of docker/addon.py](https://github.com/home-assistant/supervisor/blob/8c528f7ec54148d9847016a6c71737373e6742a5/supervisor/docker/addon.py#L121) to:
`for key, value in self.addon.schema.validate(self.addon.options).items():`

Though not sure if that's going to be the most correct answer, or if this is on purpose for security etc.

**Comment by mdegat01** [comment]
Sorry but we're not going to do this. This would definitely be a breaking change. Essentially no validation of the addon options is being done currently when the addon's config says `legacy: true`. This change would not only affect secrets, it would mean every provided option is now properly validated. Which I realize sounds like a good thing (and generally it is) but it isn't happening for addons using legacy config currently so we aren't going to start now.

Legacy means old and no longer recommended. The option is there for backwards compatibility but there are no enhancements for it anymore. You should ask the developer of that addon to update the addon and stop using it.

Also for reference `!secret` isn't a security feature. It's a sharing feature. By putting in placeholders you can share your config without accidently giving away private info. But since the actual secrets are stored in an unencrypted file in `/config` that anything with access to that folder can look it its really not about security. This `eufy_security` addon [maps /config](https://github.com/fuatakgun/eufy_security_addon/blob/main/config.json#L19) for example so it can freely read anything in `secrets.yaml` or in the `.storage` folder.

In short while it is nice that `!secret` keeps the value out of the addon UI, you really shouldn't think of it as more secure. If someone you don't trust has admin access to your HA instance (actually really any access) then you have a problem. Having `!secret` references there isn't going to stop someone from getting that value if they really want it (among other bad things).
